src/cbor.zig

```diff
@@ -318,6 +335,31 @@ pub const ArrayIterator = struct {
     }
 };
 
+/// Iterator for iterating over an indefinite length array, returned by DataItem.arrayIndef()
+pub const IndefArrayIterator = struct {
+    data: []const u8,
+    i: usize,
+    count: usize,
+
+    pub fn next(self: *@This()) ?DataItem {
+        // break marker means iterator is done
+        const T = Type.fromByte(self.data[0]);
+        if (T == Type.Break) return null;
```

Comment:
The two lines above don't reliably check for the `Break`-byte as they always check the first byte of the `data` field (index 0) which never changes.